src/IntervalTrees.jl

```diff
@@ -1018,7 +1022,7 @@ end
 
 function Base.haskey(t::InternalNode{K, V, B}, key::AbstractInterval{K}) where {K, V, B}
     i = findidx(t, key)
-    if i <= length(t) - 1 && key >= t.keys[i]
+    if i <= length(t) - 1 && !(key < t.keys[i])
```

Comment:
Please review this code in light of #36. While this is just a workaround a complete elaborate fix will needed as described in #36.